trieste/space.py

```diff
@@ -241,11 +247,55 @@ def __contains__(self, value: TensorType) -> bool | TensorType:
         return tf.reduce_all(value >= self._lower) and tf.reduce_all(value <= self._upper)
 
     def sample(self, num_samples: int) -> TensorType:
+        """
+        Sample randomly from the space.
+
+        :param num_samples: The number of points to sample from this search space.
+        :return: ``num_samples`` i.i.d. random points, sampled uniformly, and without replacement,
+            from this search space with shape ('N', 'D').
+        """
+        tf.debugging.assert_non_negative(num_samples)
+
         dim = tf.shape(self._lower)[-1]
         return tf.random.uniform(
             (num_samples, dim), minval=self._lower, maxval=self._upper, dtype=self._lower.dtype
         )
 
+    def sample_halton(self, num_samples: int, seed: int = 0) -> TensorType:
+        """
+        Sample from the space using a Halton sequence. The resulting samples are guaranteed to be
+        diverse and are reproducible by using the same choice of ``seed``.
+
+        :param num_samples: The number of points to sample from this search space.
+        :param seed: Random seed for the halton sequence
+        :return: ``num_samples`` of points, using halton sequence with shape ('N', 'D').
+        """
+        tf.random.set_seed(seed)
+        tf.debugging.assert_non_negative(num_samples)
+        if num_samples == 0:
+            return []
+        dim = tf.shape(self._lower)[-1]
+        return (self._upper - self._lower) * tfp.mcmc.sample_halton_sequence(
+            dim=dim, num_results=num_samples, dtype=self._lower.dtype, seed=seed
+        ) + self._lower
+
+    def sample_sobol(self, num_samples: int, skip: int = 0) -> TensorType:
+        """
+        Arange points using sobol sample. The resulting samples are deterministic and skipped
+        the first ``skip`` points of the sobol sequence
```

Comment:
Sample from the space using a Sobol sequence. The resulting samples are deterministic (for a given skip) and guaranteed to be diverse.